File: paper_artifacts/experiments/llr8/artifacts/sources/kimi27sglang-c/fuse_stencil_through_transient__submitted.c
#include <stdint.h>

void fuse_stencil_through_transient_fp64(double *restrict a,
                                         double *restrict out,
                                         int64_t LEN_1D,
                                         uint8_t *workspace,
                                         int64_t workspace_bytes)
{
    (void)workspace;
    (void)workspace_bytes;

    #pragma omp parallel for schedule(static) if(LEN_1D > 4096)
    for (int64_t i = 1; i < LEN_1D - 2; ++i) {
        double s0 = a[i - 1] + a[i] + a[i + 1];
        double s1 = a[i] + a[i + 1] + a[i + 2];
        out[i] = s0 * s1;
    }
}

File: paper_artifacts/experiments/llr8/artifacts/sources/oss120b-fortran-skills/fuse_stencil_through_transient__submitted.f90
! Fortran implementation of fuse_stencil_through_transient kernel
! Computes out(i) = (a(i-1) + a(i) + a(i+1)) * (a(i) + a(i+1) + a(i+2))
! for i = 2 .. LEN_1D-2 (1-based indexing)

subroutine fuse_stencil_through_transient_fp64(a, out, LEN_1D, workspace, workspace_size) bind(C, name="fuse_stencil_through_transient_fp64")
  use iso_c_binding
  implicit none
  integer(c_int64_t), value, intent(in) :: LEN_1D
  type(c_ptr), value, intent(in) :: workspace
  integer(c_int64_t), value, intent(in) :: workspace_size
  real(c_double), intent(in) :: a(LEN_1D)
  real(c_double), intent(inout) :: out(LEN_1D)
  integer(c_int64_t) :: i

  ! Parallelize the loop; each iteration is independent.
  !$omp parallel do simd schedule(static)
  do i = 2, LEN_1D-2
    out(i) = (a(i-1) + a(i) + a(i+1)) * (a(i) + a(i+1) + a(i+2))
  end do
end subroutine

